modelling/src/main/java/org/axonframework/modelling/saga/repository/AnnotatedSagaRepository.java

```diff
@@ -232,30 +236,29 @@ protected AnnotatedSaga<T> doLoadSaga(String sagaIdentifier) {
         if (entry != null) {
             T saga = entry.saga();
             resourceInjector.injectResources(saga);
-            return new AnnotatedSaga<>(sagaIdentifier, entry.associationValues(), saga, sagaModel);
+            return new AnnotatedSaga<>(sagaIdentifier, entry.associationValues(), saga, sagaModel, chainedInterceptor);
         }
         return null;
     }
 
     /**
      * Builder class to instantiate a {@link AnnotatedSagaRepository}.
      * <p>
-     * The {@link ResourceInjector} is defaulted to a {@link NoResourceInjector}.
-     * This Builder either allows directly setting a {@link SagaModel} of generic type {@code T}, or it will generate
-     * one based of the required {@code sagaType} field of type {@link Class}. Thus, either the SagaModel <b>or</b> the
-     * {@code sagaType} should be provided. All Sagas in this repository must be {@code instanceOf} this saga type.
-     * Additionally, the {@code sagaType} and {@link SagaStore} are <b>hard requirements</b> and as such should be
-     * provided.
+     * The {@link ResourceInjector} is defaulted to a {@link NoResourceInjector}. This Builder either allows directly
+     * setting a {@link SagaModel} of generic type {@code T}, or it will generate one based of the required
+     * {@code sagaType} field of type {@link Class}. Thus, either the SagaModel <b>or</b> the {@code sagaType} should be
```

Comment:
Nit: I'm missing the pointer that what applies to the `SagaModel` also applies to the `MessageHandlerInterceptorMemberChain`.

Reply:
added